### ISSUE TITLE
wrong __all__ in twisted/internet/main.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4636]
|<img alt="ivank's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| ivank reported|
|-|-|
|Trac ID|trac#4636|
|Type|defect|
|Created|2010-08-16 19:13:26Z|

change [29905] added `ReactorAlreadyInstalledError` to `__all__` in twisted/internet/main.py, but `ReactorAlreadyInstalledError` was really added to twisted/internet/error.py (which has no `__all__`)

Found by amaury in [#4476](https://github.com/twisted/twisted/issues/4476)

Attachments:

* [find_wrong_all.py](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/0d9/0d9f7dddfa136f4e8a4b8f6313f27a23618f699d/a16eb084f7b2961c9aad45c6c18069e3f6d39376.py) (1448 bytes) - added by skopchal on 2010-08-17 12:29:29Z - 
* [4636.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/0d9/0d9f7dddfa136f4e8a4b8f6313f27a23618f699d/a35b9a5f89ce8667f3ccaeeb50268cd0bb3906df.patch) (447 bytes) - added by binjured on 2011-01-22 19:01:23Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4636 4636
type__defect defect
reporter__ivank ivank
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1281986006000000 1281986006000000
changetime__1295723685000000 1295723685000000
version__None None
owner__ 
cc__amaury cc__binjured
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="skopchal's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|skopchal commented|
|-|-|

There is same problem in twisted/words/iwords.py:

```
[importing] twisted.words.iwords [Attribute Error]
        module twisted.words.iwords doesnt have attribute IChatInterface
```

I attached small script, which i wrote to check all the modules.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

The pyflakes builder will also point these out: http://buildbot.twistedmatrix.com/builders/pyflakes/builds/316/steps/pyflakes/logs/undefined

**Comment by twisted-trac** [comment]
|<img alt="binjured's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|binjured commented|
|-|-|

I couldn't find anything that imported ReactorAlreadyInstalledError from t.i.main so, err, here's a patch?

**Comment by twisted-trac** [comment]
|<img alt="binjured's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|binjured removed owner|
|-|-|

Tagged for review/reassigned IAW http://twistedmatrix.com/trac/wiki/TwistedDevelopment#SubmittingaPatch

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @exarkun|
|-|-|

Copyright date on `main.py` should be updated, and a news fragment is needed (although this strikes me as a misc).  These are trivial, so I'll add them and apply.  Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [30565]) Remove `ReactorAlreadyInstalledError` from `twisted.internet.main.__all__` since it is not available in that module.

Author: binjured
Reviewer: exarkun
Fixes: [#4636](https://github.com/twisted/twisted/issues/4636)


Remove a name from an `__all__` which was not defined in the module.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
